spark/src/main/java/org/datacleaner/spark/SparkJobLauncher.java

```diff
@@ -0,0 +1,119 @@
+import org.datacleaner.connection.Datastore;
+import org.datacleaner.job.AnalysisJob;
+import org.datacleaner.job.JaxbJobReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkJobLauncher {
+    
+    private static final Logger logger = LoggerFactory.getLogger(SparkJobLauncher.class);
+
+    private final DataCleanerConfiguration _dataCleanerConfiguration;
+
+    public SparkJobLauncher(DataCleanerConfiguration dataCleanerConfiguration) {
+        _dataCleanerConfiguration = dataCleanerConfiguration;
+    }
+
+    public SparkJobLauncher(String confXmlPath) {
+        _dataCleanerConfiguration = readDataCleanerConfiguration(new HdfsResource(confXmlPath));
+    }
+
+    public void launchJob(AnalysisJob analysisJob) {
+        final String datastoreName = analysisJob.getDatastore().getName();
+
+        final Datastore datastore = _dataCleanerConfiguration.getDatastoreCatalog().getDatastore(datastoreName);
+        if (datastore == null) {
+            throw new IllegalStateException("Datastore referred by the job (" + datastoreName
+                    + ") has not been found in the specified DataCleanerConfiguration");
+        } else {
+            JavaSparkContext sc = null;
+            try {
+                DataContext dataContext = datastore.openConnection().getDataContext(); 
+                if (dataContext instanceof CsvDataContext) {
+                    CsvDataContext csvDataContext = (CsvDataContext) dataContext;
+                    String datastoreFilePath = csvDataContext.getResource().getQualifiedPath();
+            
+                    SparkConf conf = new SparkConf().setAppName("DataCleaner-spark").setMaster("yarn-client");
+                    sc = new JavaSparkContext(conf);
+
+                    JavaRDD<String> instructionsRDD = sc.textFile(datastoreFilePath);
+                    long lineCount = instructionsRDD.count();
+        
+                    logger.info("Line count: " + lineCount);
+                }
+            } finally {
+                if (sc != null) {
+                    sc.stop();
+                }
+            }
+        }
+    }
+
+    public void launchJob(String analysisJobXmlPath) {
+        final AnalysisJob analysisJob = readAnalysisJob(new HdfsResource(analysisJobXmlPath));
+        launchJob(analysisJob);
+    }
+
+    private static DataCleanerConfiguration readDataCleanerConfiguration(HdfsResource confXmlHdfsResource) {
+        final InputStream confXmlInputStream = confXmlHdfsResource.read();
+        final JaxbConfigurationReader confReader = new JaxbConfigurationReader();
+        final DataCleanerConfiguration dataCleanerConfiguration = confReader.create(confXmlInputStream);
+
+        return dataCleanerConfiguration;
+    }
+
+    private AnalysisJob readAnalysisJob(HdfsResource analysisJobXmlHdfsResource) {
+        final InputStream analysisJobXmlInputStream = analysisJobXmlHdfsResource.read();
```

Comment:
This stream is never closed. In general I recommend using the read(Func) method to avoid unclosed resources in a guaranteed fashion.